src/hash.cr

```diff
@@ -912,6 +910,21 @@ class Hash(K, V)
     value
   end
 
+  # Sets the value of *key* to the given *value*.
+  #
+  # If a value already exists for `key`, that (old) value is returned.
+  # Otherwise the given block is invoked and its value is returned.
+  #
+  # ```
+  # h = {} of Int32 => String
+  # h.store(1, "one") { "didn't exist" } # => "didn't exist"
+  # h.store(1, "uno") { "didn't exist" } # => "one"
+  # ```
+  def store(key : K, value : V)
```

Comment:
```suggestion
  def store(key : K, value : V, &block)
```

Reply:
This is not strictly needed.

Reply:
IMO it would not bad to become mandatory.
Here this is for the API docs. I forgot to add the type signature, I edit the suggestion.

Reply:
API docs include this even if you don't do it in code. If it becomes mandatory I'll add it.

Reply:
Not the return type:
- without: `def store(key : K, value : V, &block)`
- with: `def store(key : K, value : V, &block : -> U) V | U forall U`

One will know what the method returns without reading the comments.
I only see advantages to add it.

Reply:
Yes, that's what I originally wrote. Then I looked at fetch and it has no return type or block annotation.

I understand that the type annotations make it more formal, but at the same time they make the signature more complex. It reminds me of those long Rust function signatures.

I prefer brevity and clarity. It's easy to understand how the method works by reading the example.